setup.py

```diff
@@ -101,6 +101,47 @@ def parse_arg_remove_string(argv, arg_name_equal):
 is_manylinux = environ.get('AUDITWHEEL_PLAT', None) in manylinux_tags
 
 
+def build_torch_cpp_extensions():
```

Comment:
May we break this function down into multiple functions (one for cuda/rocm extensions and another for non-cuda/rocm extensions?

Reply:
for cuda and rocm, they work the same way.
This function breaks down extensions in two categories. one for cpu-only and another for cpu+gpu extensions
we have a work item to properly handle cpp extensions. we can take this on that work item